src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexWriteOperation.cs

```diff
@@ -60,14 +68,48 @@ public class IndexWriteOperation : IndexOperationBase
             {
                 throw new IndexWriteException(e);
             }
+
+            var mapReduceIndex = index as MapReduceIndex;
+            var outputReduceToCollection = mapReduceIndex?.Definition.OutputReduceToCollection;
+            if (string.IsNullOrWhiteSpace(outputReduceToCollection) == false)
+            {
+                _outputReduceToCollectionCommand = new OutputReduceToCollectionCommand(_documentDatabase, outputReduceToCollection);
+            }
         }
 
         public override void Dispose()
         {
             try
             {
                 if (_writer != null) // TODO && _persistance._indexWriter.RamSizeInBytes() >= long.MaxValue)
-                    _writer.Commit(); // just make sure changes are flushed to disk
+                {
+                    if (_outputReduceToCollectionCommand == null)
+                    {
+                        _writer.Commit(); // just make sure changes are flushed to disk
+                    }
+                    else
+                    {
+                        using (_stats.SaveOutputDocumentsStats.Start())
```

Comment:
Note that this entire method is already measured here:
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Server/Documents/Indexes/Index.cs#L998